From 1ce292b61775cc1f49527d1081ae675880cedc2b Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:26 -0400
Subject: [ethernet] be2net: ignore VF mac address setting for the same mac

Message-id: <1410192881-30702-64-git-send-email-ivecera@redhat.com>
Patchwork-id: 92469
O-Subject: [RHEL7.1 PATCH 63/78] be2net: ignore VF mac address setting for the same mac
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit 3c31aaf340387a209b6f7036adaa8522a1cd7f18
Author: Vasundhara Volam <vasundhara.volam@emulex.com>
Date:   Fri Aug 1 17:47:31 2014 +0530

    be2net: ignore VF mac address setting for the same mac

    ndo_set_vf_mac() call may be issued for a mac-addr that is already
    active on a VF. If so, silently ignore the request.

    Signed-off-by: Vasundhara Volam <vasundhara.volam@emulex.com>
    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index f2db1df..570c8d2 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -1270,6 +1270,12 @@ static int be_set_vf_mac(struct net_device *netdev, int vf, u8 *mac)
  if (!is_valid_ether_addr(mac) || vf >= adapter->num_vfs)
   return -EINVAL;
 
+ /* Proceed further only if user provided MAC is different
+  * from active MAC
+  */
+ if (ether_addr_equal(mac, vf_cfg->mac_addr))
+  return 0;
+
  if (BEx_chip(adapter)) {
   be_cmd_pmac_del(adapter, vf_cfg->if_handle, vf_cfg->pmac_id,
     vf + 1);
-- 
1.7.1